### PR TITLE
cranelift: Created test for Function

### PR DESCRIPTION
**Pull request by antonromanov1**

### CODE REVIEW

**Comment by antonromanov1** [comment]
@cfallin please take a look.

**Comment by antonromanov1** [comment]
@afonso360 please take a look.

**Comment by cfallin** [comment]
Hi @antonromanov1 -- thanks for the PR! Could you fill in the PR description with some text that explains this PR?

In particular: I'm curious, is there any particular reason you found that indicates we need a two-line test for `Function::new()` to create an empty function, and nothing else? The `Function::new()` function is pretty well covered already by all of our other tests that create functions.

To be clear, I'm not at all opposed to "simple" tests if there's a good reason for them, and especially as a nice way for new folks to get started with our codebase; we should just have a good reason for them :-)

**Comment by antonromanov1** [comment]
@cfallin my test also covers method `Function::is_leaf` (moreover it asserts the returning value).

File: cranelift/codegen/src/ir/function.rs [hunk 1]
```diff
@@ -495,3 +495,16 @@ impl<'a> Iterator for InstOffsetIter<'a> {
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn new_function() {
+        let func = Function::new();
+        let block = func.dfg.make_block();
+        assert!(func.is_leaf());
+        assert_eq!(func.is_block_basic(block), Ok(()));
```
Comment (bjorn3) [inline cranelift/codegen/src/ir/function.rs:508]: This function won't pass the verifier as there is no entry block. The first block for which an instruction is inserted is in principle the entry block. All blocks need to end with a terminator instruction.
